### PR TITLE
Fix RadioButtonGroup defaultValue (of 0) & event.target.value (in case of number value) issues

### PR DESCRIPTION
**Pull request by vaishnavyogesh**
#### What does this PR do?
1) Handle the defaultValue of 0 being as one of the options value
2) Retrieve correct selected value in onChange using event.value instead of event.target.value (this breaks the code when the value was a number but event.target.value gives it as a string and the component breaks)

#### Where should the reviewer start?
RadioButtonGroup Component

#### How should this be manually tested?
A new story of RadioButtonGroup is created under the name `Simple` to manually see the issue fixes.

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Any background context you want to provide?
A new story of RadioButtonGroup is created under the name `Simple` to manually see the issue fixes.

#### What are the relevant issues?

#6751 

#### Screenshots (if appropriate)
image

#### Do the grommet docs need to be updated?
Yes, to mention that one needs to get value from event.value instead of event.target.value

#### Is this change backwards compatible or is it a breaking change?
I don't think it's a breaking change, since the previous method of retrieving value in onChange using event.target.value still works fine.

### CODE REVIEW

**Comment by vaishnavyogesh** [comment]
Resolves issue #6751

**Comment by vaishnavyogesh** [comment]
Hey team, could anyone guide me please, why is the first check is failing?

ci/circleci: bundlesize — Your tests failed on CircleCI

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js [hunk 1]
```diff
@@ -48,7 +48,8 @@ const RadioButtonGroup = forwardRef(
     const [value, setValue] = formContext.useFormInput({
       name,
       value: valueProp,
-      initialValue: defaultValue || '',
+      // 0 could be one of the options value
+      initialValue: defaultValue === 0 ? defaultValue : defaultValue || '',
```
Comment (ericsoderberghp) [inline src/js/components/RadioButtonGroup/RadioButtonGroup.js:52]:
```suggestion
      initialValue: defaultValue ?? '',
```

**Comment by jcfilben** [comment]
> Hey team, could anyone guide me please, why is the first check is failing?
>
> ci/circleci: bundlesize — Your tests failed on CircleCI

Hey @vaishnavyogesh we have a bundle size check just to make sure that a change isn't causing the bundle size to greatly increase. In this case we are just slightly above our bundle size, which is no problem we just need to update the max bundle size specified in the `package.json` file. 

In the `package.json` file change the `bundlesize.maxSize` to be 160 kB instead of 159 kB
```javascript
"bundlesize": [
    {
      "path": "./dist/grommet.min.js",
      "maxSize": "160 kB"
    }
  ],
```

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js (hunk 1 above)
Comment (vaishnavyogesh) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js]: Fixed!

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js [hunk 2]
```diff
@@ -101,7 +102,16 @@ const RadioButtonGroup = forwardRef(
       }, 1);
     };
 
+    const onRadioButtonChange = (event, optionValue) => {
+      setValue(optionValue);
+      event.persist(); // extract from React synthetic event pool
+      const adjustedEvent = event;
+      adjustedEvent.value = optionValue;
+      if (onChange) onChange(adjustedEvent);
```
[the remark this answers is not in the document]
Comment (vaishnavyogesh) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js]: Fixed!

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js [hunk 3]
```diff
@@ -101,7 +102,16 @@ const RadioButtonGroup = forwardRef(
       }, 1);
     };
 
+    const onRadioButtonChange = (event, optionValue) => {
+      setValue(optionValue);
+      event.persist(); // extract from React synthetic event pool
```
[the remark this answers is not in the document]
Comment (vaishnavyogesh) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js]: Done adding required comment!

**Comment by vaishnavyogesh** [comment]
Hey @jcfilben could you tell me, what's the first check is trying to tell now?
`1 change must be accepted as baseline`
is there anything to be done from my end for the first two tests to pass now/

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js [hunk 4]
```diff
@@ -101,7 +101,20 @@ const RadioButtonGroup = forwardRef(
...
 
+    const onRadioButtonChange = (event, optionValue) => {
+      setValue(optionValue);
+      if (onChange) {
+        event.persist(); // extract from React synthetic event pool
+        
+        // event.target.value gives value as a string which needs to be
+        // manually typecasted according to the type of original option value.
+        // return the original option value attached with the event.
+        onChange({...event, value: optionValue});
```
Comment (ericsoderberghp) [inline src/js/components/RadioButtonGroup/RadioButtonGroup.js:112]: I seem to recall an issue with using `...event` like this in the past. Unfortunately, I cannot remember the details. But, I think we should keep the pattern of:
```
const adjustedEvent = event;
adjustedEvent.value = optionValue;
```

**Comment by vaishnavyogesh** [comment]
Hey @ericsoderberghp is it compulsory to have run through all the tests on a git commit than just testing the component that we've changed even after knowing that the changes won't affect other components?

File: src/js/components/RadioButtonGroup/RadioButtonGroup.js (hunk 4 above)
Comment (vaishnavyogesh) [reply src/js/components/RadioButtonGroup/RadioButtonGroup.js]: Fixed!

**Comment by vaishnavyogesh** [comment]
Hey @ericsoderberghp, @jcfilben any updates on this?

**Comment by jcfilben** [comment]
> Hey @ericsoderberghp, @jcfilben any updates on this?

Sorry for the delay in review on this! Thanks for making the changes and reaching back out to us to check in.

> Hey @jcfilben could you tell me, what's the first check is trying to tell now?
> 1 change must be accepted as baseline
> is there anything to be done from my end for the first two tests to pass now/

This check just lets us know that one of the visual snapshots of our storybook stories has been changed. This is expected since this PR makes a change to the RadioButtonGroup simple story.

> Hey @ericsoderberghp is it compulsory to have run through all the tests on a git commit than just testing the component that we've changed even after knowing that the changes won't affect other components?

Good question! We like to have all the tests run on git commit because there are some components that are built with other components in grommet. For example the DateInput component is built with the Drop and Calendar components. So if we made a change to Drop it could have the potential to affect other components.

**Review by jcfilben — APPROVED** [review]
Looks good!

**Comment by vaishnavyogesh** [comment]
Got it!

**Review by ericsoderberghp — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**